### PR TITLE
Feature/time trigger ui

### PR DESCRIPTION
**Pull request by jozefbakus**
Time trigger fragment with simple validation (will not let you save workflow with incorrect cron expression).
Related to backend implementation: #56

### CODE REVIEW

File: src/main/resources/ui/view/sensor/time.fragment.xml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  -->
+
+<core:FragmentDefinition xmlns="sap.m" xmlns:core="sap.ui.core">
+    <Label text="Will run"/>
+    <Text text="{parts: ['/workflow/sensor/properties/settings/variables/cronExpression'],
+                 formatter: 'Formatters.cronFormatter'}" />
+    <Label text="Cron Expression"/>
+    <Input type="Text"
+           value="{/workflow/sensor/properties/settings/variables/cronExpression}"
+           placeholder="* * * * * *"
```
Comment (kevinwallimann) [inline src/main/resources/ui/view/sensor/time.fragment.xml]: Isn't this the invalid expression?

File: src/main/resources/ui/controller/UpsertWorkflow.controller.js [hunk 2]
```diff
@@ -104,14 +104,27 @@ sap.ui.define([
...
+					WorkflowRepository.createWorkflow(workflow);
+				}
+				this.myNavBack();
 			}
-			this.myNavBack();
+		},
+
+		_isWorkflowValid: function (workflow) {
+			try {
+				cronstrue.toString(workflow.sensor.properties.settings.variables.cronExpression);
```
Comment (kevinwallimann) [inline src/main/resources/ui/controller/UpsertWorkflow.controller.js]: You could add a custom validation here for that specific case that cronstrue checks wrongly (one of day of month and day of week must be `?`)

Comment (kevinwallimann) [reply src/main/resources/ui/controller/UpsertWorkflow.controller.js]: In fact, please do

**Review by kevinwallimann — CHANGES_REQUESTED** [review]

File: src/main/resources/ui/controller/UpsertWorkflow.controller.js [hunk 3]
```diff
@@ -104,14 +104,27 @@ sap.ui.define([
...
-				WorkflowRepository.createWorkflow(workflow);
+			if(this._isWorkflowValid(this._model.getProperty("/workflow"))){
+				let isEdit = this._model.getProperty("/isEdit");
+				let workflow = this.getWorkflowToSave();
+				if(isEdit) {
+					WorkflowRepository.updateWorkflow(workflow);
+				} else {
+					WorkflowRepository.createWorkflow(workflow);
+				}
+				this.myNavBack();
```
Comment (Zejnilovic) [inline src/main/resources/ui/controller/UpsertWorkflow.controller.js]: The tabbing seems extensive. How about spaces and a max of 4?

File: src/main/resources/ui/controller/UpsertWorkflow.controller.js [hunk 4]
```diff
@@ -186,7 +199,8 @@ sap.ui.define([
 
 		sensorTypes: [
 			{name: "Absa-Kafka", fragment: "absaKafka", controller: "AbsaKafkaController"},
-			{name: "Kafka", fragment: "kafka", controller: "KafkaController"}
+			{name: "Kafka", fragment: "kafka", controller: "KafkaController"},
+			{name: "Time", fragment: "time", controller: "TimeController"}
```
Comment (Zejnilovic) [inline src/main/resources/ui/controller/UpsertWorkflow.controller.js:201]: This should be slowly moved to configuration

File: src/main/resources/ui/controller/UpsertWorkflow.controller.js (hunk 2 above)
Comment (Zejnilovic) [reply src/main/resources/ui/controller/UpsertWorkflow.controller.js]: If you do it as in Enceladus and let user only select what is possible, then you can skip the validations. 

BTW Why don't you use the output of the `toString` function to show to the users, so they know what time they have set up. This is one of the really convenient features for the user

File: src/main/resources/ui/controller/UpsertWorkflow.controller.js (hunk 4 above)
Comment (jozefbakus) [reply src/main/resources/ui/controller/UpsertWorkflow.controller.js:201]: Will be fixed here: #60

File: src/main/resources/ui/controller/UpsertWorkflow.controller.js (hunk 3 above)
Comment (jozefbakus) [reply src/main/resources/ui/controller/UpsertWorkflow.controller.js]: Will be fixed here: #60

File: src/main/resources/ui/controller/UpsertWorkflow.controller.js (hunk 2 above)
Comment (jozefbakus) [reply src/main/resources/ui/controller/UpsertWorkflow.controller.js]: Kevin - yes, fixed
Sasa - 1. Yes in Angular UI. 2. Yes, we do.

File: src/main/resources/ui/view/sensor/time.fragment.xml (hunk 1 above)
Comment (jozefbakus) [reply src/main/resources/ui/view/sensor/time.fragment.xml]: Yes, fixed

File: src/main/resources/ui/controller/sensor/TimeController.js [hunk 5]
```diff
@@ -29,8 +29,15 @@ class TimeController {
         let cronExpressionPath = "/workflow/sensor/properties/settings/variables/cronExpression";
         let cronExpression = this._model.getProperty(cronExpressionPath);
         if(!cronExpression) {
-            this._model.setProperty(cronExpressionPath, "0 0 * * * *")
+            this._model.setProperty(cronExpressionPath, "0 0 0 ? * * *")
```
Comment (kevinwallimann) [inline src/main/resources/ui/controller/sensor/TimeController.js:32]: Did you change this intentionally from a 6-character to a 7-character cron expression?

**Review by kevinwallimann — APPROVED** [review]

File: src/main/resources/ui/controller/sensor/TimeController.js (hunk 5 above)
Comment (jozefbakus) [reply src/main/resources/ui/controller/sensor/TimeController.js:32]: Yes, 7th character is optional, but the expression is still valid.

**Pull request merged**